espGarden/main/include/http_server.h

```diff
@@ -4,157 +4,81 @@
-    ctx_data->status_flag = true;
+    rx_buf[rx] = 0;
+    ESP_LOGI(HTTP_LOG_TAG, "Received: %s", rx_buf);
+    
+    // Process the command here (e.g., water the garden for an hour)
+    // process_watering_command(system_state, rx_buf);
 
-    /* Respond */
-    const char resp[] = "Adder POST Response";
+    // Send a receive packet
+    const char resp[] = "rec";
     httpd_resp_send(req, resp, HTTPD_RESP_USE_STRLEN);
+    ESP_LOGI(HTTP_LOG_TAG, "Sent Receipt Confimation: %s", resp);
+
     return ESP_OK;
 }
 
-/* Custom URI handler function */
-esp_err_t my_uri_handler(httpd_req_t *req)
-{
-    set_cors_headers(req);    
-    // Recv, Process and Send
-    const char resp[] = "Custom URI Response";
-    httpd_resp_send(req, resp, HTTPD_RESP_USE_STRLEN);
+/* Handler function for sending state data */
+esp_err_t tx_handler(httpd_req_t *req) {
+    char tx_buff[RX_TX_BUFF_LEN]; // Buffer for sending data
 
-    // On success
+    // Test DATA we are sending for now
+    const char * test_data = "DATA";
+    snprintf(tx_buff, sizeof(tx_buff), "%s", test_data);
+
+    // Send the data
+    httpd_resp_set_type(req, "application/json");
+    httpd_resp_send(req, tx_buff, HTTPD_RESP_USE_STRLEN);
+    ESP_LOGI(HTTP_LOG_TAG, "Sent: %s", tx_buff);
     return ESP_OK;
 }
 
-/* Function to start the webserver */
-httpd_handle_t start_webserver(void)
-{
+/* Function to start the web server */
+httpd_handle_t start_webserver(void) {
     httpd_config_t config = HTTPD_DEFAULT_CONFIG();
     httpd_handle_t server = NULL;
 
+    // One URI for receiving, one for sending
     if (httpd_start(&server, &config) == ESP_OK) {
-        httpd_uri_t uri_get = {
-            .uri = "/uri/get",
-            .method = HTTP_GET,
-            .handler = get_handler,
-            .user_ctx = NULL
-        };
-
-        httpd_uri_t uri_post = {
-            .uri = "/uri/post",
-            .method = HTTP_POST,
-            .handler = post_handler,
-            .user_ctx = NULL
-        };
-        httpd_uri_t uri_adder_post = {
-            .uri = "/uri/post/adder",
+        httpd_uri_t rx_uri = {
+            .uri = "/esp_rx",
             .method = HTTP_POST,
-            .handler = adder_post_handler,
+            .handler = rx_handler,
             .user_ctx = NULL
         };
 
-        httpd_register_uri_handler(server, &uri_get);
-        httpd_register_uri_handler(server, &uri_post);
-
-        // Example of registering custom URI handler
-        httpd_uri_t my_uri = {
-            .uri = "/my_uri/path/xyz",
+        httpd_uri_t tx_uri = {
+            .uri = "/esp_tx",
             .method = HTTP_GET,
-            .handler = my_uri_handler,
+            .handler = tx_handler,
             .user_ctx = NULL
         };
 
-        if (httpd_register_uri_handler(server, &my_uri) != ESP_OK) {
-            // Handle failure to register handler
-        }
+        httpd_register_uri_handler(server, &rx_uri);
+        httpd_register_uri_handler(server, &tx_uri);
```

Comment:
this is good i thinks needed if we don't have a conventional /uri/

Reply:
rx_uri and tx_uri are just variable names here. The actual endpoints are ip/esp_rx for messages going to the esp such as watering commands and ip/esp_tx for commands going to the server such as system state status updates or moisture sensor data in future